src/solver/variable/economy/shortTermStorage.h

```diff
@@ -100,37 +100,55 @@ struct VCardShortTermStorageEnergy
             case 1:
                 return "PSP_open_withdrawal";
             case 2:
-                return "PSP_closed_injection";
+                return "PSP_open_level";
```

Comment:
It would be nice if we could avoid having this function, use a loop instead. But I know there's a problem with `const char*` causing segfaults...

We should consider switching to `std::string` eventually.

Reply:
I have a problem to do that.

In the case of the pollutants, the returned **std::string**  has an address that cannot be corrupted : it is an element of a constant standard container (an **std::map**). This container is built at compile time and requires no computation.

In the current case, we also need a constant container to hold the strings returned internally inside `static const char* Caption(/*...*/ index)`, but its content cannot be computed at compile time, as it is **const**.

I leave the code as it is until we find a solution to this.

Reply:
ok, I switched to std::string as a return type of all **Caption(...)** static member functions